soh/src/code/z_parameter.c

```diff
@@ -1415,15 +1437,18 @@ void Inventory_SwapAgeEquipment(void) {
         }
 
         // When becoming adult, remove swordless flag since we'll get master sword
+        // (Unless Master Sword is shuffled)
         // Only in rando to keep swordless link bugs in vanilla
-        if (gSaveContext.n64ddFlag) {
+        if (gSaveContext.n64ddFlag && !Randomizer_GetSettingValue(RSK_SHUFFLE_MASTER_SWORD)) {
             gSaveContext.infTable[29] &= ~1;
         }
 
         gSaveContext.childEquips.equipment = gSaveContext.equips.equipment;
 
         if (gSaveContext.adultEquips.buttonItems[0] == ITEM_NONE) {
```

Comment:
The current reason for adult c-button items not saving when without the master sword is because it re-enters this condition. May want to expand this so that it checks that adult equips is initialised.

e.g. 
```
if (gSaveContext.adultEquips.buttonItems[0] == ITEM_NONE && 
    !(gSaveContext.n64ddFlag && Randomizer_GetSettingValue(RSK_SHUFFLE_MASTER_SWORD) && gSaveContext.adultEquips.equipment != 0))
```

though there is probably a better way for checking if you haven't been adult yet and this looks kinda messy